diff --git a/configure b/configure
index 8f9fb04115..a70c24378d 100755
--- a/configure
+++ b/configure
@@ -1996,8 +1996,6 @@ EXTERNAL_LIBRARY_GPL_LIST="
     libdvdread
     librubberband
     libvidstab
-    libx264
-    libx265
     libxavs
     libxavs2
     libxvid
@@ -2010,6 +2008,8 @@ EXTERNAL_LIBRARY_NONFREE_LIST="
 "
 
 EXTERNAL_LIBRARY_VERSION3_LIST="
+    libx264
+    libx265
     gmp
     libaribb24
     liblensfun
